compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -457,24 +457,56 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
             }
             sym::patchable_function_entry => {
                 codegen_fn_attrs.patchable_function_entry = attr.meta_item_list().and_then(|l| {
-                    let mut prefix = 0;
-                    let mut entry = 0;
+                    let mut prefix = None;
+                    let mut entry = None;
                     for item in l {
-                        if let Some((sym, lit)) = item.name_value_literal() {
```

Comment:
Why did we switch off `name_value_literal()`? It looks like you've re-implemented it by first extracting out the symbol manually (even if `name_value_literal()` doesn't work for some reason, you should consider `symbol_or_empty()` rather than the first two steps). I do see that you can no longer directly call it on the `item` and would need to use `if let Some((sym, lit)) = item.meta_item().and_then(|mi| mi.name_value_literal()) {`, but it seems like it'd be a lot shorter?

Reply:
Thanks for suggesting symbol_or_empty! I found name_or_empty, which was probably what you meant and it allowed me to get rid of that ugly span parsing.
I also took the oppurtunity add better error diagnostics, when parsing fails.